### PR TITLE
Avoid runtime crashes by removing undefined behavior.

### PR DESCRIPTION
**Pull request by mario-vera**
From pthread docs:
"The effect of calling pthread_getspecific() or pthread_setspecific() with a key value not obtained from pthread_key_create() or after key has been deleted with pthread_key_delete() is undefined."

On some Linux flavors this results in getspecific returning uninitialized pointer with results in runtime crashes after dereferencing invalid memory.

Also NULL format for fprintf is invalid and results on crashes when calling CFShow.

### CODE REVIEW

File: CoreFoundation/Base.subproj/CFPlatform.c [hunk 1]
```diff
@@ -628,6 +628,7 @@ static void *__CFTSDGetSpecific() {
 #if DEPLOYMENT_TARGET_MACOSX || DEPLOYMENT_TARGET_EMBEDDED || DEPLOYMENT_TARGET_EMBEDDED_MINI
     return pthread_getspecific(__CFTSDIndexKey);
 #elif DEPLOYMENT_TARGET_LINUX
+    __CFTSDInitialize();
```
Comment (parkera) [inline CoreFoundation/Base.subproj/CFPlatform.c]: I think this actually belongs here:

https://github.com/apple/swift-corelibs-foundation/blob/ac7f8030ad381f0f3d97359f574a6c97a6cf66e1/CoreFoundation/Base.subproj/CFRuntime.c#L1080

where we can probably just remove the requirement for `DEPLOYMENT_RUNTIME_SWIFT` and make it `DEPLOYMENT_TARGET_LINUX`.

**Comment by millenomi** [comment]
I think you have the commits for your other PR merged into this one. Can you separate them?

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Pull request merged**